components/registry-facade/pkg/registry/blob.go

```diff
@@ -175,12 +175,14 @@ func (bh *blobHandler) getBlob(w http.ResponseWriter, r *http.Request) {
 		defer bufPool.Put(bp)
 
 		var n int64
+		var serr error
 		err = wait.PollImmediateWithContext(ctx, 100*time.Millisecond, time.Minute, func(context.Context) (done bool, err error) {
-			n, err = io.CopyBuffer(w, rc, *bp)
-			if err == nil {
+			n, serr = io.CopyBuffer(w, rc, *bp)
+			if serr == nil {
 				return true, nil
 			}
-			if errors.Is(err, syscall.ECONNRESET) || errors.Is(err, syscall.EPIPE) {
+			if errors.Is(serr, syscall.ECONNRESET) || errors.Is(serr, syscall.EPIPE) {
+				log.WithField("blobSource", src.Name()).WithField("baseRef", bh.Spec.BaseRef).WithError(serr).Warn("retry get blob because of error")
 				return false, nil
 			}
 			return true, err
```

Comment:
Should this return `serr` now as well?

```suggestion
			return true, serr
```

Reply:
:100:

Reply:
👍 
In that case, is there a need to add `serr`? `PollImmediateWithContext` will return the error that was returned inside the retry function, so think it's fine to use `err`?

Reply:
Yes, there still is because PollImmediateWithContext could return original error, like that:
https://github.com/gitpod-io/gitpod/blob/9d2e9b1a2dec673623a8b31923a28954710109ae/components/ws-manager/pkg/manager/manager.go#L381-L384

Reply:
:100: